portal/src/main/webapp/js/lib/oql/test-oql-parser.js

```diff
@@ -11,7 +11,16 @@ var testAlterationEquality = function(expected, given) {
 			for (var key in exp_alt) {
 				if (exp_alt.hasOwnProperty(key)) {
 					same_keys = same_keys && giv_alt.hasOwnProperty(key);
-					same_values = same_values && (giv_alt[key] === exp_alt[key]);
+					var giv_val = giv_alt[key];
+					var exp_val = exp_alt[key];
+					if (typeof giv_val !== typeof exp_val) {
+					    same_values = false;
+					    break;
+					} else if (typeof giv_val === "object") {
+					    same_values = same_values && testAlterationEquality(giv_val, exp_val);
```

Comment:
this is to test that the object is equal, i.e. same keys and same values, and the recursion is to do this if some of the values are themselves objects, which can't be checked with `===`.